File: Sample/java/src/main/java/net/nend/sample/java/nativead/customviews/MarqueeView.java
package net.nend.sample.java.nativead.customviews;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.HorizontalScrollView;

public class MarqueeView extends HorizontalScrollView {

    public MarqueeView(Context context) {
        super(context);
    }

    public MarqueeView(Context context, AttributeSet attrs) {
        super(context, attrs);
        requestDisallowInterceptTouchEvent(true);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return super.onInterceptTouchEvent(ev);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return false;
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/nativead/MarqueeView.kt
package net.nend.sample.kotlin.nativead

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.HorizontalScrollView

class MarqueeView : HorizontalScrollView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        requestDisallowInterceptTouchEvent(true)
    }

    override fun onTouchEvent(ev: MotionEvent) = false
}